packages/jest-message-util/src/index.ts

```diff
@@ -358,10 +358,58 @@ export const formatStackTrace = (
 };
 
 type FailedResults = Array<{
+  /** Stringified version of the error */
   content: string;
+  /** Details related to the failure */
+  failureDetails: unknown;
+  /** Execution result */
   result: TestResult.AssertionResult;
 }>;
 
+function formatErrorStack(
+  errorOrStack: Error | string,
+  config: StackTraceConfig,
+  options: StackTraceOptions,
+  testPath?: string,
+): string {
+  // The stack of new Error('message') contains both the message and the stack,
+  // thus we need to sanitize and clean it for proper display using separateMessageFromStack.
+  const sourceStack =
+    typeof errorOrStack === 'string' ? errorOrStack : errorOrStack.stack || '';
+  let {message, stack} = separateMessageFromStack(sourceStack);
+  stack = options.noStackTrace
+    ? ''
+    : `${STACK_TRACE_COLOR(
+        formatStackTrace(stack, config, options, testPath),
+      )}\n`;
+
+  message = checkForCommonEnvironmentErrors(message);
+  message = indentAllLines(message);
+
+  let cause = '';
+  if (
+    typeof errorOrStack !== 'string' &&
+    'cause' in errorOrStack &&
+    errorOrStack.cause
+  ) {
+    if (
+      typeof errorOrStack.cause === 'string' ||
```

Comment:
this is always `false` I think? you check `typeof errorOrStack !== 'string'` in the outer

Reply:
Except for the root call with rootErrorOrStack